### PR TITLE
Improvements to the INT8 GEMM portion of the code for Power

### PR DESCRIPTION
**Pull request by ChipKerchner**
These are changes to improve GEMM portion of the code for Power.

There are 2 main code changes : 
1) Changing a function to a template parameter so that operations that add/sub zero are eliminated at compile time.  Plus reuse a vector that has the mask instead of rebuilding each time.
2) Add processing 16 columns at a time in MlasGemmQuantCopyPackB8x8 - this should reduce potential page faults by a factor of 4 and also be faster.
3) Unroll MlasQgemmStoreVectorMMA and vectorize other variables.

### CODE REVIEW

**Comment by ChipKerchner** [comment]
Questions about the lint warnings.  Do all lines have to be less than 120 characters?  And for statements that span multiple lines, how are they written?

**Comment by yuslepukhin** [comment]
Typically, we employ clangformat. I use a visual cue in the IDE and break them manually, then run the formatter again.

**Comment by ChipKerchner** [comment]
Which `style` should I be using for clang-format?

**Comment by yuslepukhin** [comment]
> Which `style` should I be using for clang-format? microsoft?
> 
> It seems the formatter wants to change a lot of code that I did not alter.

Your editor should pick this up automatically.
https://github.com/microsoft/onnxruntime/blob/main/.clang-format

**Comment by ChipKerchner** [comment]
I use `vim` as my editor.  I'm not sure it will pickup lint formatting.

**Comment by yuslepukhin** [comment]
> I use `vim` as my editor. I'm not sure it will pickup lint formatting.

Most of the failures about extra space. Lots of editors show non-visible characters.

File: onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp [hunk 1]
```diff
@@ -521,31 +543,133 @@ MlasGemmQuantCopyPackA8x8(
...
     size_t ldb,
     size_t CountN,
     size_t CountK,
-    int32_t* ColumnSumBuffer,
-    bool BIsSigned
+    int32_t* ColumnSumBuffer
     )
 {
     const uint8_t BitFlipValue = (BIsSigned ? 0x80 : 0);
+    ORT_UNUSED_PARAMETER(BitFlipValue);
```
Comment (edgchen1) [inline onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp:558]: why add `ORT_UNUSED_PARAMETER` here? isn't BitFlipValue used just below in the vmask assignment?

File: onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp [hunk 2]
```diff
@@ -67,16 +67,15 @@ MlasGemmQuantFixupZeroPointB<MLAS_GEMM_QUANT_KERNEL_POWER10>(
...
     const uint8_t* A,
     size_t lda,
     size_t CountM,
     size_t CountK,
-    int32_t* RowSumBuffer,
-    bool AIsSigned
+    int32_t* RowSumBuffer
     )
 {
     const uint8_t Flip = (AIsSigned ? 0 : 0x80);
```
Comment (edgchen1) [inline onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp:81]:
```suggestion
    constexpr uint8_t Flip = (AIsSigned ? 0 : 0x80);
```

File: onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp [hunk 3]
```diff
@@ -521,31 +543,133 @@ MlasGemmQuantCopyPackA8x8(
...
     const uint8_t* B,
     size_t ldb,
     size_t CountN,
     size_t CountK,
-    int32_t* ColumnSumBuffer,
-    bool BIsSigned
+    int32_t* ColumnSumBuffer
     )
 {
     const uint8_t BitFlipValue = (BIsSigned ? 0x80 : 0);
```
Comment (edgchen1) [inline onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp:557]:
```suggestion
    constexpr uint8_t BitFlipValue = (BIsSigned ? 0x80 : 0);
```

**Review by edgchen1 — COMMENTED** [review]
can you share performance measurements to show how much improvement there is?

**Comment by ChipKerchner** [comment]
I'm seeing about a 2.6-4X improvement for PackB

File: onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp (hunk 1 above)
Comment (ChipKerchner) [reply onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp:558]: Some compilers still emit an unused variable - even though it is not true.

Comment (yuslepukhin) [reply onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp:558]: Suggest using `[[maybe_unused]]` attribute.

**Comment by ChipKerchner** [comment]
Can we move forward with this PR?

**Review by yufenglee — APPROVED** [review]

**Comment by yufenglee** [comment]
> Can we merge this?

Thanks @ChipKerchner !

**Pull request merged**